pkgs/development/libraries/tpm2-tss/default.nix

```diff
@@ -70,10 +70,11 @@ stdenv.mkDerivation rec {
       --replace 'git describe --tags --always --dirty' 'echo "${version}"'
   '';
 
-  configureFlags = lib.optionals (stdenv.buildPlatform == stdenv.hostPlatform) [
-    "--enable-unit"
-    "--enable-integration"
-  ];
+  preConfigure = lib.optionals (stdenv.buildPlatform == stdenv.hostPlatform) ''
+    if [ ! -z $doCheck ]; then
+      configureFlagsArray+=('--enable-unit' '--enable-integration')
```

Comment:
This would be a good use case for makeDerivation as a function (having a hard time to find the description, but grep for 'mkDerivation.*final'

Reply:
This should use finalAttrs.finalPackage.doCheck as a second condition instead.